voila/configuration.py

```diff
@@ -52,30 +54,62 @@ class VoilaConfiguration(traitlets.config.Configurable):
     )
     strip_sources = Bool(True, config=True, help="Strip sources from rendered html")
 
-    file_whitelist = List(
+    file_allowlist = List(
         Unicode(),
         [r".*\.(png|jpg|gif|svg)"],
         config=True,
         help=r"""
     List of regular expressions for controlling which static files are served.
-    All files that are served should at least match 1 whitelist rule, and no blacklist rule
-    Example: --VoilaConfiguration.file_whitelist="['.*\.(png|jpg|gif|svg)', 'public.*']"
+    All files that are served should at least match 1 allowlist rule, and no denylist rule
+    Example: --VoilaConfiguration.file_allowlist="['.*\.(png|jpg|gif|svg)', 'public.*']"
     """,
     )
 
-    file_blacklist = List(
+    file_whitelist = List(
+        Unicode(),
+        [r".*\.(png|jpg|gif|svg)"],
+        config=True,
+        help="""Deprecated, use `file_allowlist`""",
+    )
+
+    @validate("file_whitelist")
+    def _valid_file_whitelist(self, proposal):
+        warn(
+            "Deprecated, use VoilaConfiguration.file_allowlist instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return proposal["value"]
```

Comment:
May I suggest another approach for this?

```python
@property
def file_whitelist(self):
   warn(
        "Deprecated, use VoilaConfiguration.file_allowlist instead.",
        DeprecationWarning,
        stacklevel=2,
    )
   return self.file_allowlist 


@file_whitelist.setter
def file_whitelist(self, value):
   warn(
        "Deprecated, use VoilaConfiguration.file_allowlist instead.",
        DeprecationWarning,
        stacklevel=2,
    )
   self.file_allowlist = value
```

That way we can get rid of the

```python
warnings.filterwarnings("default", category=DeprecationWarning, module="traitlets")
```

?

Reply:
My suggestion applies if we consider going from a traitlet to a property is not backward incompatible

Reply:
By doing so, users can not set the allow/deny list from the CLI or the `voila.json` file